### PR TITLE
Add EIP: Cross-Contract Hierarchical NFT

### PR DESCRIPTION
**Pull request by minkyn**

### CODE REVIEW

File: EIPS/eip-nft_hierarchy.md [hunk 1]
```diff
@@ -0,0 +1,247 @@
+---
+eip: <TBA>
```
Comment (abcoathup) [inline EIPS/eip-nft_hierarchy.md:2]:
```suggestion
eip: 7510
```
Assigning next sequential EIP/ERC number.
EIP/ERC numbering changed to sequential from 7500 and is no longer the PR number.

Please also update the filename.

File: EIPS/eip-7510.md [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+
+In addition, we didn't include `child` relation in the interface. An original NFT exists before its derivative NFTs. Therefore we know what parent tokens to include when minting derivative NFTs, but we wouldn't know the children tokens when minting the original NFT. If we have to record the children, that means whenever we mint a derivative NFT, we need to call on its original NFT to add it as a child. However, those two NFTs may belong to different contracts and thus require different write permissions, making it impossible to combine the two operations into a single transaction in practice. As a result, we decide to only record the `parent` relation from the derivative NFTs.
+
+## Backwards Compatibility
+
+No backwards compatibility issues found.
+
+## Test Cases
+
+Test cases available at: [ERC7510.test.ts](../assets/eip-7510/test/ERC7510.test.ts):
```
Comment (xinbenlv) [inline EIPS/eip-7510.md:66]:
```suggestion
Test cases available at: [`ERC7510.test.ts`](../assets/eip-7510/test/ERC7510.test.ts):
```

File: EIPS/eip-7510.md [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+        }
+        delete _parentTokens[tokenId];
+    }
+
+}
+```
+
+## Security Considerations
+
+No security considerations found.
```
Comment (xinbenlv) [inline EIPS/eip-7510.md:243]: Specify the case when parent NFT is burned or changed

**Comment by SamWilsn** [comment]
I am closing this pull request because we are in the process of separating EIPs and ERCs into distinct repositories. Unfortunately, as far as we are aware, GitHub does not provide any tools to ease this migration, so every pull request will need to be re-opened manually.

As this is a PR to create / modify an ERC, I will kindly ask you to redirect this to the new repository at [ethereum/ERCs](https://github.com/ethereum/ERCs). **We have prepared [a guide](https://github.com/ethereum/ERCs/issues/8) to help with the process.**

If there is relevant history here, please link to this PR from the new pull request.

On behalf of the EIP Editors, I apologize for this inconvenience.

**Pull request closed without merging**